### PR TITLE
use helm values to set chaos-daemon capabilities

### LINKED ISSUE #3917: Chaos daemon requires NET_RAW capability on OpenShift
## Bug Report

**What version of Kubernetes are you using?**
Azure OpenShift
```
→ oc version
Client Version: 4.12.0-202208031327
Kustomize Version: v4.5.4
Server Version: 4.10.40
Kubernetes Version: v1.23.12+7566c4d
→ k version --short
Flag --short has been deprecated, and will be removed in the future. The --short output will become the default.
Client Version: v1.26.1
Kustomize Version: v4.5.7
Server Version: v1.23.12+7566c4d
WARNING: version difference between client (1.26) and server (1.23) exceeds the supported minor version skew of +/-1
```

**What version of Chaos Mesh are you using?**
```
kubectl exec -n chaos-testing chaos-controller-manager-55c4f86574-bdffb -- /usr/local/bin/chaos-controller-manager -version
Controller manager Version: version.Info{GitVersion:"v2.0.5", GitCommit:"e4994c9e1cf61285c0991a6e4c71be850edb6dfc", BuildDate:"2021-11-25T10:17:51Z", GoVersion:"go1.15.11", Compiler:"gc", Platform:"linux/amd64"}
```

**What did you do? / Minimal Reproducible Example**
```
→ helm upgrade chaos-mesh chaos-mesh/chaos-mesh \
		--install \
		--create-namespace \
		--namespace=chaos-testing \
		--version=2.0.5 \
		--set chaosDaemon.privileged=false \
		--set chaosDaemon.runtime=crio \
		--set chaosDaemon.socketPath=/var/run/crio/crio.sock \
		--wait
→ oc adm policy add-scc-to-user privileged --namespace chaos-testing --serviceaccount chaos-daemon
```

Applying a NetworkChaos to our database workload to simulate a network partition:
```
apiVersion: chaos-mesh.org/v1alpha1
kind: NetworkChaos
metadata:
  name: network-partition-innodb-cluster
spec:
  action: partition
  mode: all
  selector:
    labelSelectors:
      'app.kubernetes.io/instance': 'mysql-ha-sample'
      'app.kubernetes.io/component': 'database'
  direction: both
  target:
    selector:
      labelSelectors:
        'app.kubernetes.io/instance': 'mysql-ha-sample'
        'app.kubernetes.io/component': 'database'
    mode: all
```

**What did you expect to see?**

Our pods should be unable to communicate with each other, and the database would report a loss of quorum.

**What did you see instead?**

The pods are able to communicate, so our test that expects the loss of quorum fails because quorum is maintained.

We only spot-checked chaos-daemon logs and saw no errors (but realizing now we may not have looked at the right node). However chaos-controller-manager logs contained the below:

`→ k -n chaos-testing logs chaos-controller-manager-55c4f86574-bdffb`
```
2023-02-07T19:18:40.628Z	INFO	iptable	attempting to set ip table chains	{"containerName": "mysqld-exporter", "containerID": "cri-o://a68f7aa7bcc21068ddf1bd48ae4e4a131a528d2b809f8578b5aae5fb13e69b19"}
2023-02-07T19:18:40.632Z	ERROR	iptable	error while setting ip tables chains for container mysqld-exporter, id cri-o://a68f7aa7bcc21068ddf1bd48ae4e4a131a528d2b809f8578b5aae5fb13e69b19	{"error": "rpc error: code = Unknown desc = error code: exit status 3, msg: iptables v1.8.2 (legacy): can't initialize iptables table `filter': Permission denied (you must be root)\nPerhaps iptables or your kernel needs to be upgraded.\n"}
... [stacktrace]
2023-02-07T19:18:40.632Z	ERROR	podnetworkchaos	fail to set iptables	{"error": "unable to set ip tables chains for pod mysql-ha-sample-0"}
```

Based on these error messages, I found an OpenShift bug report: https://bugzilla.redhat.com/show_bug.cgi?id=1895032
... which indicates that the container may also need `NET_RAW` capability, because apparently that capability was recently dropped by OpenShift.

We tested that hypothesis by patching the daemonset:
`kubectl --namespace chaos-testing patch daemonset chaos-daemon --type=json \
			--patch '[{"op": "add", "path": "/spec/template/spec/containers/0/securityContext/capabilities/add/-", "value":"NET_RAW"}]'`

And then our test passed: the database lost quorum as expected, and the iptables errors were no longer produced in the chaos cotroller logs.

**Output of chaosctl**
Please let me know if you need this, as it will take some effort to understand what is needed.

CC: @achasveachas @swati-nair

### PR DESCRIPTION
**Pull request by nioshield**
### What problem does this PR solve?
Flexible configuration of capabilities is expected in some test scenarios

### What's changed and how it works?
Replace the chaos-daemon-daemonset.yaml capabilities configuration with template variables

### Related changes
fixes #3917 

- [ ] This change also requires further updates to the [website](https://github.com/chaos-mesh/website) (e.g. docs)
- [ ] This change also requires further updates to the `UI interface`
- Need to **cheery-pick to release branches**
  - [ ] release-2.5
  - [ ] release-2.4

### Checklist

CHANGELOG

- [ ] I have updated the `CHANGELOG.md`
- [ ] I have labeled this PR with "no-need-update-changelog"

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] No code
- [ ] Manual test (add steps below)

Side effects

- [ ] **Breaking backward compatibility**

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

**Comment by cwen0** [comment]
@nioshield this pr can't fix https://github.com/chaos-mesh/chaos-mesh/issues/3917, but is is useful. We can merge it.

**Review by STRRL — COMMENTED** [review]
rest LGTM

File: helm/chaos-mesh/values.yaml [hunk 1]
```diff
@@ -169,16 +169,16 @@ chaosDaemon:
...
-  ##     - MKNOD
-  ##     - SYS_CHROOT
-  ##     - SYS_ADMIN
-  ##     - KILL
-  ##     # CAP_IPC_LOCK is used to lock memory
-  ##     - IPC_LOCK
+  capabilities:
+    add:
+      - SYS_PTRACE
+      - NET_ADMIN
```
[the remark this answers is not in the document]
Comment (nioshield) [reply helm/chaos-mesh/values.yaml:175]: I think so. Let me add it

**Pull request merged**